src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapper.java

```diff
@@ -571,34 +583,30 @@ protected void parseCreateField(ParseContext context, int dimension, SpaceType s
     }
 
     // Verify mapping and return true if it is a "faiss" Index using "sq" encoder of type "fp16"
-    protected boolean isFaissSQfp16(KNNMethodContext knnMethodContext) {
-
-        // KNNMethodContext shouldn't be null
-        if (Objects.isNull(knnMethodContext)) {
+    protected boolean isFaissSQfp16(MethodComponentContext methodComponentContext) {
+        if (Objects.isNull(methodComponentContext)) {
             return false;
         }
 
-        // engine should be faiss
-        if (!FAISS_NAME.equals(knnMethodContext.getKnnEngine().getName())) {
+        if (methodComponentContext.getParameters().size() == 0) {
             return false;
         }
 
-        // Should have Method Component Parameters
-        if (knnMethodContext.getMethodComponentContext().getParameters().size() == 0) {
-            return false;
-        }
-        Map<String, Object> methodComponentParams = knnMethodContext.getMethodComponentContext().getParameters();
+        Map<String, Object> methodComponentParams = methodComponentContext.getParameters();
 
         // The method component parameters should have an encoder
         if (!methodComponentParams.containsKey(METHOD_ENCODER_PARAMETER)) {
             return false;
         }
 
-        MethodComponentContext methodComponentContext = (MethodComponentContext) methodComponentParams.get(METHOD_ENCODER_PARAMETER);
+        MethodComponentContext encoderMethodComponentContext = (MethodComponentContext) methodComponentParams.get(METHOD_ENCODER_PARAMETER);
 
         // returns true if encoder name is "sq" and type is "fp16"
-        return ENCODER_SQ.equals(methodComponentContext.getName())
-            && FAISS_SQ_ENCODER_FP16.equals(methodComponentContext.getParameters().getOrDefault(FAISS_SQ_TYPE, FAISS_SQ_ENCODER_FP16));
+        return ENCODER_SQ.equals(encoderMethodComponentContext.getName())
+            && FAISS_SQ_ENCODER_FP16.equals(
+                encoderMethodComponentContext.getParameters().getOrDefault(FAISS_SQ_TYPE, FAISS_SQ_ENCODER_FP16)
```

Comment:
Can we just match strictly or can the encoder type for sq not be set?

Reply:
Yeah, right now as we are only supporting `fp16` for `sq` encoder it is set as default.